src/dbus_api/udev.rs

```diff
@@ -37,11 +37,12 @@ impl DbusUdevHandler {
         let udev_event = self.receiver.recv().await.ok_or_else(|| {
             StratisError::Error("Channel from udev handler to D-Bus handler was shut".to_string())
         })?;
-        let mut mutex_lock = self.dbus_context.engine.lock().await;
-        let optional_pool_info = mutex_lock.handle_event(&udev_event);
+        let mut lock = self.dbus_context.engine.write().await;
```

Comment:
We don't want to use the `lock` macro in `async` methods as it blocks.

Reply:
Maybe it would be helpful to make two locking macros, ```block_lock``` for the blocking lock, and ```yield_lock``` for the yielding lock, like here. I think they could be used in a completely identical fashion as
```block_lock!(self.dbus_context.engine, write)``` and ```yield_lock!(self.dbus_context.engine, write)```. I think even though it's visual only, i.e., no type distinction, it would assist with code comprehension a good deal.

Reply:
Per our discussion, we're going to keep it as is.